### DISCUSSION TITLE
Zugriff auf die Influxdb Datenbank / Error

### DISCUSSION THREAD

**Discussion by prawnhitter** [solectrus/solectrus#1875] [Q&A]
Hallo zusammen,
Hallo @ledermann,

irgendwas mache ich falsch. Ich versuche gerade auf die InfluxDB Datenbank zuzugreifen um mir die Werte auszulesen und eine andere Auswertung für mich zusammenzustellen mit Grafana. Dazu muss ich die DB einrichten.

- Query Language: Flux
- URL: http://192.168.xxx.xxx:8086
- Auth: Basic Auth
- ->User: admin
- ->Pwd: ExAmPl3PA55W0rD
- Influxbd Org: solectrus
- Token: my-super-secret-admin-token
- Bucket: my-solectrus-bucket

Wenn ich jetzt dies abspeichere bekomme ich eine Fehlermeldung
"Post "http://192.168.xxx.xxx:8086/api/v2/query?org=solectrus": dial tcp 192.168.xxx.xxx:8086: connect: connection refused error reading buckets"

Wo liegt mein Fehler hierbei? Kann mir jemand helfen?

**Comment by ledermann** [comment]
InfluxDB stellt eine Admin-Oberfläche bereit, die über den Browser bedient wird. Dazu ist ein Login mit Benutzername und Passwort erforderlich. Die in der SOLECTRUS-Installationsanleitung aufgeführten [Standard-Zugangsdaten](https://github.com/solectrus/hosting/blob/main/guide/raspberry-pi/.env#L39-L44) sind dabei, wie Du richtig angegeben hast `admin/ExAmPl3PA55W0rD`.

Damit das aber funktioniert, muss der Port `8086` nach außen hin geöffnet werden, was standardmäßig **nicht** der Fall ist. Ich vermute, das ist hier die Ursache. Details hier:
https://github.com/solectrus/hosting/blob/main/guide/raspberry-pi/docker-compose.yml#L48-L50

**Reply by prawnhitter** [reply]
okay, das war schon mal der Erste Fehler scheinbar. Nach der Aktivierung der TCP Port Einstellungen im influxdb Container ist nun die oben beschriebene Fehlermeldung weg und nun kommt

"unauthorized: unauthorized access error reading buckets"

Aber eigentlich müsste doch sowohl der token als auch das "basic Auth" alles abdecken.

**Reply by ledermann** [reply]
Hm, offenbar benutzt Du nicht die Web-Oberfläche von InfluxDB, denn dazu braucht es keinen Token und kein `BasicAuth`. Womit versuchst Du denn auf InfluxDB zuzugreifen? Mich irritiert insbesondere `BasicAuth`. Und: Klappt denn der Login über den Browser?

Ansonsten könnte es am Token liegen. Du kannst über die Admin-Oberfläche von InfluxDB ein spezifisches Custom-Token erstellen, das genau den Zugriff auf den benötigten Bucket zulässt.

**Reply by prawnhitter** [reply]
Hi, sorry für den späten Reply, bin bisher leider nicht in der Nähe meiner Rechnerstruktur.

Ich komme auf die Weboberfläche, aber wenn ich in "Grafana" versuche es zu verbinden, kommt eben der Fehler.

Ich werde sobald ich wieder am Rechner bin es nochmal probieren einen separaten Token zu erzeugen.
